templates/media/media-with-message.php

```diff
@@ -0,0 +1,173 @@
+
+	</script>
+	<span class="primary rtm-media-msg-upload-button rtmedia-upload-media-link" 
+		id="rtm_show_upload_ui" title="Upload Media">
+		<i class="dashicons dashicons-upload rtmicon"></i>
+			<?php echo esc_html__( 'Upload Media File', 'buddypress-media' ); ?>
+	</span>
+	<div id="rtm-media-gallery-uploader" class="rtm-media-gallery-uploader">
+		<?php
+		rtmedia_uploader(
+			[
+				'is_up_shortcode' => false,
+				'allow_anonymous' => true,
+				'privacy_enabled' => false,
+			]
+		);
+			?>
+	</div>
+	<input type="hidden" id="rtm_bpm_uploaded_media" name="rtm_bpm_uploaded_media" />
+	<?php
+}
+
+/**
+ * Add additional parameter media ID in BuddyPress message SEND MESSAGE process.
+ * And Insert data into MEDIA_META table.
+ *
+ * @param object $message Getting parameter response when sending message.
+ **/
+function rtm_add_message_media_params( $message ) {
+	$insert_media_object  = new RTDBModel( 'rtm_media_meta' );
+	$message->media_array = sanitize_text_field( filter_input( INPUT_POST, 'rtm_bpm_uploaded_media' ) );
+	$media                = explode( ',', $message->media_array );
+	if ( ! empty( $media ) && null !== $media ) {
+		foreach ( $media as $media_id ) {
+			$insert_media_object->insert(
+				[
+					'media_id'   => $media_id,
+					// Adding meta data into custom meta table.
+					'meta_key'   => 'rtm-bp-message-media', // phpcs:ignore
+					// Adding meta data into custom meta table.
+					'meta_value' => $message->id, // phpcs:ignore
+				]
+			);
+		}
+	}
+	?>
+	<script>
+		jQuery( '#msg-success-bp-msg-media' ).hide();
+		jQuery( '.rtm-media-msg-upload-button' ).attr( 'id', 'rtm_show_upload_ui' );
+		jQuery( '.rtm-media-msg-upload-button' ).html( '' );
+		jQuery( '.rtm-media-msg-upload-button' ).html( 
+			jQuery( '<i>',
+				{ class: 'dashicons dashicons-upload rtmicon' }
+			) );
+		jQuery( '#rtm_show_upload_ui' ).append( 'Upload Media File' );
+	</script>
+	<?php
+}
+
+/**
+ * As a result show attached media with message.
+ *
+ * @var object $get_data_object            Object for having DB actions.
+ * @var array  $media_result               Getting media_ids attached with messages.
+ * @var array  $url                        Partitions of current URL to redirect to perticular media.
+ * @var mixed  $rtm_gallary_list_filter    Filter for Media gallary listing.
+ * @var mixed  $rtm_change_alt_text_filter Filter for changing alter text of an image.
+ **/
+function show_rtm_bp_msg_media() {
+	$get_data_object            = new RTDBModel( 'rtm_media_meta' );
+	$media_result               = $get_data_object->get( [ 'meta_value' => bp_get_the_thread_message_id() ] );  // phpcs:ignore
+	$url                        = explode( 'messages/', sanitize_text_field( wp_unslash( filter_input( INPUT_SERVER, 'REQUEST_URI' ) ) ) );
+	$rtm_gallary_list_filter    = apply_filters( 'rtmedia_gallery_list_item_a_class', 'rtmedia-list-item-a' );
+	$rtm_change_alt_text_filter = apply_filters( 'rtmc_change_alt_text', $alt_text, $rtmedia_media );
```

Comment:
I told you to add inline documentation for these new filters.

Reply:
Filters are already there in rtmedia-filters... I have just re-used them. 
So I have mention why I have used them in Function Doc Comment above.

/**
 *As a result show attached media with message.
 *
 *@var object $get_data_object            Object for having DB actions.
 *@var array  $media_result               Getting media_ids attached with messages.
 *@var array  $url                        Partitions of current URL to redirect to perticular media.
 *@var mixed  $rtm_gallary_list_filter    Filter for Media gallary listing.
 *@var mixed  $rtm_change_alt_text_filter Filter for changing alter text of an image.
 **/